### PR TITLE
Extract Qt atomic operation wrappers into separate header file 

### PR DESCRIPTION
**Pull request by uklotzde**
Based on #4304 that should be merged first.

### CODE REVIEW

**Comment by daschuer** [comment]
#4304 is merged now. I wonder why the changes are still in the diff here.

File: src/util/compatibility/qatomic.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#include <QAtomicInteger>
+#include <QAtomicPointer>
+
+template<typename T>
+inline T atomicLoadAcquire(const QAtomicInteger<T>& atomicInt) {
+    // TODO: QBasicAtomicInteger<T>::load() is deprecated and should be
+    // replaced with QBasicAtomicInteger<T>::loadRelaxed() However, the
+    // proposed alternative has just been introduced in Qt 5.14. Until the
+    // minimum required Qt version of Mixxx is increased, we need a version
+    // check here
```
Comment (Be-ing) [inline src/util/compatibility/qatomic.h:12]: Does this comment really need to be repeated almost verbatim 6 times? I think one comment at the top of the file would be sufficient.

**Comment by Be-ing** [comment]
> #4304 is merged now. I wonder why the changes are still in the diff here.

GitHub will show the first commit until the branch is rebased on main.

**Comment by uklotzde** [comment]
Rebased and cleaned up the comments.

**Review by Be-ing — APPROVED** [review]
Waiting for CI

**Pull request merged**